chia/wallet/wallet_coin_store.py

```diff
@@ -62,58 +62,65 @@ async def create(cls, wrapper: DBWrapper):
         await self.db_connection.execute("CREATE INDEX IF NOT EXISTS wallet_id on coin_record(wallet_id)")
 
         await self.db_connection.commit()
-        self.coin_record_cache = dict()
-        self.coin_wallet_record_cache = {}
-        all_coins = await self.get_all_coins()
-        for coin_record in all_coins:
-            self.coin_record_cache[coin_record.coin.name()] = coin_record
-
-        self.wallet_cache_lock = asyncio.Lock()
+        self.coin_record_cache = {}
+        self.unspent_coin_wallet_cache = {}
+        await self.rebuild_wallet_cache()
         return self
 
     async def _clear_database(self):
         cursor = await self.db_connection.execute("DELETE FROM coin_record")
         await cursor.close()
         await self.db_connection.commit()
 
+    async def rebuild_wallet_cache(self):
+        # First update all coins that were reorged, then re-add coin_records
+        all_coins = await self.get_all_coins()
+        self.unspent_coin_wallet_cache = {}
+        self.coin_record_cache = {}
+        for coin_record in all_coins:
```

Comment:
If we do need to iterate over all coins, I think this would be a lot more efficient to iterate over the sqlite cursor here. To stream them in from the database.